src/model/events-model.js

```diff
@@ -1,44 +1,75 @@
 import Observable from '../framework/observable';
-import { generateEvent } from '../mock/events';
-import { getRandomInteger } from '../utils/common';
-
-const generateEvents = () => {
-  const key = getRandomInteger(1, 4);
-
-  switch (key) {
-    case 1:
-      return Array.from({length: 4}, generateEvent);
-    case 2:
-      return Array.from({length: 7}, generateEvent);
-    case 3:
-      return Array.from({length: 11}, generateEvent);
-    default:
-      return [];
-  }
-};
+import { UpdateType } from '../const';
 
 export default class EventsModel extends Observable {
-  #events = generateEvents();
+  #eventsApiService = null;
+  #events = [];
+  #destinations = [];
+  #offers = [];
+
+  constructor(eventsApiService) {
+    super();
+    this.#eventsApiService = eventsApiService;
+  }
 
   get events() {
     return this.#events;
   }
 
+  get destinations() {
+    return this.#destinations;
+  }
+
+  get offers() {
+    return this.#offers;
+  }
+
+  init = async () => {
+    try {
+      const events = await this.#eventsApiService.events;
+      this.#events = events.map(this.#adaptEventToClient);
+    } catch(err) {
+      this.#events = [];
+    }
+
+    try {
+      const destinations = await this.#eventsApiService.destinations;
+      this.#destinations = destinations.map(this.#adaptDestinationToClient);
+    } catch(err) {
+      this.#destinations = [];
+    }
+
+    try {
+      const offers = await this.#eventsApiService.offers;
+      this.#offers = offers.map(this.#adaptOffersToClient);
+    } catch(err) {
+      this.#offers = [];
+    }
+
+    this._notify(UpdateType.INIT);
+  };
 
-  updateEvent = (updateType, update) => {
+  updateEvent = async (updateType, update) => {
     const index = this.#events.findIndex((event) => event.id === update.id);
 
     if (index === -1) {
       throw new Error('Невозможно обновить несуществующее событие');
     }
 
-    this.#events = [
-      ...this.#events.slice(0, index),
-      update,
-      ...this.#events.slice(index + 1),
-    ];
+    try {
+      const response = await this.#eventsApiService.updateEvent(update);
+      const updatedEvent = this.#adaptEventToClient(response);
 
-    this._notify(updateType, update);
+      this.#events = [
+        ...this.#events.slice(0, index),
+        updatedEvent,
+        ...this.#events.slice(index + 1),
+      ];
+
+      this._notify(updateType, updatedEvent);
+    } catch(err) {
+      throw new Error('Can\'t update event');
```

Comment:
Всегда надо стараться ошибку выкидывать наиболее информативно чтобы твои коллеги быстрее понимали в чем проблема.
```suggestion
      throw new Error(`Can't update event. Update ${update}. Error: ${err}`);
```

Reply:
Ready